src/Traits/ConditionTrait.php

```diff
@@ -159,7 +159,7 @@ public function isMet($value, ?string $optionType = null): bool
 
                     break;
                 case ConditionComparatorEnum::EQUAL:
-                    $result = $result ? $val == $target : false;
+                    $result = $result ? $val === $target : false;
```

Comment:
We should not use `==` anymore. Should this check be weekly typed?

Reply:
I can't think of a reason to have it weakly typed.